### ISSUE TITLE
Model ID does not update

### ISSUE THREAD

**Issue by gmeik** [ptvoinfo/zigbee-configurable-firmware#178]
The Model ID remains 'ptvo.switch', also after entering a custom Model ID.
I flash the CC2530 using an Arduino with CCLoader software (after the CCDebugger was fried - delivery time of a new one is quite long). After flashing, I can operate digital outputs without problems. Using custom MQTT commands, I can also change PWM levels. So the device seems to have been flashed properly. But because the Model ID doesn't update after reprogramming/repairing, the custom converter for Zigbee2MQTT is not activated.
Let me know if I need to supply more information. Or if the CCLoader approach is the likely cause of this trouble...

**Comment by ptvoinfo** [comment]
@gmeik You should delete device in Z2M, restart Z2M and rejoin your device again. Z2M should display the new ModelId.

**Comment by gmeik** [comment]
I tried deleting the device from Z2M, restart Z2M and rejoin. It still shows up as ptvo.switch.
I checked the .bin file and the 'ptvo.switch' text does not even show up, and my own model ID does show up. So for now I'll assume that programming with CCLoader didn't work correctly, and wait for the new CCDebugger.

**Comment by ptvoinfo** [comment]
You can try to read the firmware back from the broken device and compare with the flashed HEX. The flasher uses a simple checksum checking method. It may fail in some rare cases.

**Comment by phalox** [comment]
I seem to be experiencing the same issue:

![image](https://github.com/ptvoinfo/zigbee-configurable-firmware/assets/3983738/e9a3e646-dd6a-4528-a955-576544b89696)

![image](https://github.com/ptvoinfo/zigbee-configurable-firmware/assets/3983738/00917de2-e765-4c4a-bc34-d3c0ab5bc69f)

I've deleted the device in Z2M, and restarted Z2M. None of it seems to fix the problem. When I make changes to the firmware (e.g. add an input), they do show up in Z2M.

I've also let it validate that the hex file is correct, and that seems to be the case:

![image](https://github.com/ptvoinfo/zigbee-configurable-firmware/assets/3983738/4ecb163e-6a7a-46ba-8bb4-7b98c4b03393)


I bumped into one more thing.... When exporting the converter and opening the file, I still see the old name (ptvo.switch).

![image](https://github.com/ptvoinfo/zigbee-configurable-firmware/assets/3983738/cffdaf27-91e6-4110-b7d5-608719707fa7)

Could it be that this is part of the issue?
